Code/Tools/FBuild/FBuildCore/Graph/ObjectNode.cpp

```diff
@@ -1407,7 +1407,64 @@ bool ObjectNode::BuildArgs( const Job * job, Args & fullArgs, Pass pass, bool us
                 continue;
             }
         }
-				
+
```

Comment:
I agree 4/4.  Let me know if you need help with anything - thanks!

On Sun, Jun 12, 2016 at 11:07 AM, ffulin <notifications@github.com> wrote:

> In Code/Tools/FBuild/FBuildCore/Graph/ObjectNode.cpp
> <https://github.com/fastbuild/fastbuild/pull/97#discussion_r66724912>:
>
> > @@ -1407,7 +1407,64 @@ bool ObjectNode::BuildArgs( const Job * job, Args & fullArgs, Pass pass, bool us
> >                  continue;
> >              }
> >          }
> > -				
> > +
>
> This is a good fix, but it's a little unfortunate that:
> a) we perform this logic redundantly for every obj file (potentially
> hundreds or thousands of times in a build). Of particular note is that
> EnsurePathExists is a call into the kernel.
> b) we do it in BuildArgs, which isn't really a function that should have
> side-effects
>
> Ideally, we'd have this logic performed once by the ObjectList or Library
> that owns the ObjectNodes, so we'd only have to do it once. Sadly, I don't
> think there's a good way to do that right now. I'll need to give it some
> more thought.
>
> I think I can integrate this as-is though, and look into options for
> reducing the overhead.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/fastbuild/fastbuild/pull/97/files/5701cde887ed5556a16f344f5daa1afa2bed5e86#r66724912>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADV5qDznQNgm0TGKK7M1JH3h1JZC_0m4ks5qLErYgaJpZM4IYa1J>
> .
>